toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -27,8 +27,14 @@ class GoogleBigQueryDataSource(ToucanDataSource):
     )
 
 
-# TODO - check bad return type
-def _define_type(value) -> str:
+BigQueryParam = Union[bigquery.ScalarQueryParameter, bigquery.ArrayQueryParameter]
+
+
+# NOTE: This does not play nicely with dates. They're a bit tricky
+# though, as we'd have to try and parse dates from strings to
+# determine if something is a date or not. Until then, we can just
+# use a cast. eg: SELECT * FROM table WHERE STRING(date_col) IN UNNEST({{my_dates}})
```

Comment:
Thanks for this comment :+1: 
However, it does seem to have a DATE (and even a TIMESTAMP) type available: https://googleapis.dev/python/bigquery/latest/query.html#google.cloud.bigquery.query.ScalarQueryParameterType
Can't we use them?

Reply:
Sure we could use them, but to me the problem lies with the format in which we receive the query parameters from the frontend: since we receive them as JSON, dates will be strings. Parameters are provided as a `{param_name: value}` dict, without any metadata about their type. How do we determine if a string is a date or a regular string ? Should we try to parse a date from a pre-defined list of date formats and if none of them works, consider the param to be a string ? This could be tricky, as we might have false positives, like a `"202001"` string representing an ID being interpreted as january 2020

Reply:
Good point!
We should be able to receive parameters as "real dates": we added a layer, in our HTTP API, that serializes and de-serializes dates as `{$date: <timestamp>}`. (Inspired by Mongo's extended JSON specification)

Reply:
Aaaah nice! So it should be "auto-handled" by #595